Documentation/contributor-guide/features.md

```diff
@@ -0,0 +1,60 @@
+# Features 
+
+This document provides an overview of etcd features and general guidelines for adding and removing features. The project maintainers can override the guidelines per the need of the project following the project governance.
+
+The etcd features fall into three stages, experimental, stable, and unsafe.
+
+## Experimental
+
+Any new feature should be added as an experimental feature. An experimental feature is characterized as below:
+- Might be buggy due to lack of user testing. Enabling the feature may not work as expected.
+- Support for such a feature may be dropped at any time without notice
+- Disabled by default when added initially
+
+### Minimum development requirements
+
+Experimental feature related development work is expected to be high quality, similar to any development work for the project. A related PR should meet the following criteria,
+- Associated with an issue providing a clear need for such a feature
+- Any configuration flags related to the implementation of the feature are prefixed with `experimental` e.g. `--experimental-feature-name`
+- Any variable names related to the implementation of the feature are prefixed with `Experimental` e.g. `ExperimentalFeatureName`
```

Comment:
I think this only holds true for external configuration in `embed.Server` and is not propagated deeper. I think it is a bit of overkill.